hack/test-driver

```diff
@@ -44,58 +51,111 @@ if [ "$DRIVER" != "docker" ]; then
       if [ "$firstNode" = "0" ]; then
         createFlags="$createFlags --append"
       fi
-      (
-        set -x
-        ${BUILDX_CMD} create ${createFlags} \
-          --name="${builderName}" \
-          --node="${builderName}-${platform/\//-}" \
-          --driver="${DRIVER}" \
-          --driver-opt="${driverOpt}" \
-          --platform="${platform}"
-      )
+      buildxCmd create ${createFlags} \
+        --name="${builderName}" \
+        --node="${builderName}-${platform/\//-}" \
+        --driver="${DRIVER}" \
+        --driver-opt="${driverOpt}" \
+        --platform="${platform}"
       firstNode=0
     done
   else
     createFlags=""
     if [ -f "$BUILDKIT_CFG" ]; then
       createFlags="$createFlags --config=${BUILDKIT_CFG}"
     fi
-    (
-      set -x
-      ${BUILDX_CMD} create ${createFlags} \
-        --name="${builderName}" \
-        --driver="${DRIVER}" \
-        --driver-opt="${driverOpt}" \
-        --platform="${PLATFORMS}"
-    )
+    buildxCmd create ${createFlags} \
+      --name="${builderName}" \
+      --driver="${DRIVER}" \
+      --driver-opt="${driverOpt}" \
+      --platform="${PLATFORMS}"
   fi
 fi
 
+function buildOutput {
+  local name=$1
+  if [ "$DRIVER" != "docker" ]; then
+    if [ "${MULTI_NODE}" = "1" ]; then
+      echo "type=cacheonly"
+    else
+      echo "type=oci,dest=${context}/${name}.tar"
+    fi
+  else
+    echo "type=docker,name=${name}"
+  fi
+}
+
 # multi-platform not supported by docker driver
 buildPlatformFlag=
+bakePlatformFlag=
 if [ "$DRIVER" != "docker" ]; then
   buildPlatformFlag=--platform="${PLATFORMS}"
+  bakePlatformFlag=--set="*.platform=${PLATFORMS}"
 fi
 
-set -x
-
 # inspect and bootstrap
-${BUILDX_CMD} inspect --bootstrap --builder="${builderName}"
+buildxCmd inspect --bootstrap --builder="${builderName}"
 
 # create dockerfile
 cat > "${dockerfile}" <<EOL
 FROM busybox as build
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
 RUN echo "I am running on \$BUILDPLATFORM, building for \$TARGETPLATFORM" > /log
-FROM busybox
+
+FROM busybox AS log
 COPY --from=build /log /log
 RUN cat /log
 RUN uname -a
+
+FROM busybox AS hello
+RUN echo hello > /hello
+
+FROM scratch
+COPY --from=log /log /log
+COPY --from=hello /hello /hello
 EOL
 
 # build
-${BUILDX_CMD} build ${buildPlatformFlag} \
-  --output="type=cacheonly" \
+buildxCmd build ${buildPlatformFlag} \
+  --output="$(buildOutput buildx-test-build)" \
   --builder="${builderName}" \
+  --metadata-file="${context}/metadata-build.json" \
   "${context}"
+jq '.' "${context}/metadata-build.json"
```

Comment:
Shouldn't this try some actual key?

Reply:
I'm using `.` filter just to pretty print: https://github.com/docker/buildx/runs/5139711248?check_suite_focus=true#step:9:143
No validation there. Maybe we could add some though.